### PR TITLE
Lecture 1 setup and product dao

### PR DESCRIPTION
**Pull request by nika-doroshkevich**

### CODE REVIEW

File: src/main/java/com/es/phoneshop/model/product/ArrayListProductDao.java [hunk 1]
```diff
@@ -1,25 +1,132 @@
...
-        throw new RuntimeException("Not implemented");
+        Lock writeLock = rwLock.writeLock();
+        writeLock.lock();
+
+        try {
+            var productId = product.getId();
+            if (productId != null) {
+                var foundOptionalProduct = products.stream()
+                        .filter(p -> productId.equals(p.getId()))
+                        .findAny();
```
Comment (AnnaKrikun) [inline src/main/java/com/es/phoneshop/model/product/ArrayListProductDao.java]: Can you reuse getById() method here?

File: src/main/java/com/es/phoneshop/model/product/ArrayListProductDao.java [hunk 2]
```diff
@@ -1,25 +1,132 @@
...
 import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+import java.util.stream.Collectors;
 
 public class ArrayListProductDao implements ProductDao {
+
+    private long maxId;
```
Comment (AnnaKrikun) [inline src/main/java/com/es/phoneshop/model/product/ArrayListProductDao.java:17]: Where is initial value set?

Comment (nika-doroshkevich) [reply src/main/java/com/es/phoneshop/model/product/ArrayListProductDao.java:17]: long is a primitive type, so it stores 0.

File: src/main/java/com/es/phoneshop/model/product/ArrayListProductDao.java (hunk 1 above)
Comment (nika-doroshkevich) [reply src/main/java/com/es/phoneshop/model/product/ArrayListProductDao.java]: Yep, I must have missed it.

**Review by AnnaKrikun — APPROVED** [review]

**Pull request merged**